### PR TITLE
Fix Sensei Learning Mode template and blocks broken in site editor after onboarding

### LINKED ISSUE #85038: Learning Mode template intermittently broken for Atomic sites
### Steps to Reproduce
1. Go through the Sensei onboarding flow on WordPress.com as a new user.
2. From Sensei Home, click _Customize your theme_. (You may need to verify your email address first if you haven't already.)
3. Switch your Learning Mode template to Modern or Video (may also happen with other templates, but I reproduced on these).
4. Click on Templates > Lesson (Learning Mode - Video).
5. See errors in preview window.

NOTE: This issue is intermittent for me and I've only been able to reproduce on a WordPress.com site. If you're not able to reproduce, try leaving the site editor and then coming back in. Or switch to a different Learning Mode template first.

### What I Expected
No block errors and the template to display.

### What Happened Instead
I think this error is relevant as it does not appear when the template loads correctly:

Screenshot 2023-12-05 at 2 58 27 PM

### Context / Source
Related https://github.com/Automattic/sensei/pull/7317

### PR DESCRIPTION
**Pull request by Imran92**
Related to https://github.com/Automattic/wp-calypso/issues/85038

## Proposed Changes

This issue was tricky to reproduce and fix. After noticing the template being broken, the first response from muscle memory was to deactivate Gutenberg. And it fixed the issue. Neat! So it's a clear conflict with Gutenberg! Easy peasy.

Well, it wasn't. After trying a whole bunch of things, I could not reproduce it locally, either with Gutenberg on or off. So maybe it happens with Gutenberg only on the Atomic sites! Created a bunch of atomic sites with Sensei, Sensei-Pro, and Gutenberg activated. Nope, still couldn't reproduce it. Saw a bunch of jetpack-initiated calls in the stack. So could it be a difference in Jetpack settings being different? Matched everything there is to match. Nope, nothing.

So as Sherlock Holmes once said, _"When you have eliminated all which is impossible, then whatever remains, however improbable, must be the truth"_.

So it's not Sensei, it's not Atomic, the fault is in our ~~stars~~ flow, I thought. But where in it _Horatio_? Okay, let's see if it's only a block theme issue. So I switched to Astra on a site I'm having this problem in, and this issue wasn't there anymore! Wah, so a block theme issue definitely, I thought. Just to be sure, switched to another block theme, aaaand... nope! The issue isn't there! So a Course theme issue? Switched back to the Course theme, and lo and behold, it's not happening on the Course theme either, not anymore!

So all I needed was to switch the theme and get back after the first time the site gets created using the onboarding flow? To test my theory, I created a bunch of sites using the onboarding flow and found the problem of broken templates every time, and after a theme switch and back, it was solved in all of them.

So maybe the way we were activating the Course theme during the onboarding was causing the issue, maybe it wasn't getting properly updated. So after creating the site in the flow, I added a call to switch the theme using the theme-switch API, and it was solved. Also had to add the delays, otherwise, I was seeing some discrepancies here and there.

## Testing Instructions

1. Create a Sensei site completing the whole flow after hitting `http://calypso.localhost:3000/setup/sensei`
2. Now you can check the LM template from a few different places. Mainly
    - From SenseiLMS -> Home -> Customize your theme
    - From Appearance -> Editor -> Templates -> <Your current LM template> `(check the image below)`
    - From SenseiLMS -> Settings -> Appearance -> Customize button on the template picture on hover
    - From Appearance -> Customize -> Use Site Editor -> Templates -> <Your current LM template>    
Screenshot 2023-12-09 at 2 17 59 AM
3. The LM template should not be broken, try to get out and in the editor a few times
4. Now switch to the other (pro) LM templates, go to the editor, and check them just like default one

#### Before
Screenshot 2023-12-09 at 1 36 22 AM

#### After
Screenshot 2023-12-09 at 1 37 37 AM

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by donnapep** [comment]
@Imran92 I'm having trouble testing this locally because clicking anywhere takes me to my `wpcomstaging.com` site and doesn't leave me in `http://calypso.localhost:3000/`. How could I get around this?

**Comment by Imran92** [comment]
> @Imran92 I'm having trouble testing this locally because clicking anywhere takes me to my `wpcomstaging.com` site and doesn't leave me in `http://calypso.localhost:3000/`. How could I get around this?

Thanks for taking a look at it Donna! Is it only happening in this branch or also in the trunk? Can you kindly try in some different scenarios like in a browser where you are already proxied and logged in with a wpcom account and see if this issue persists?

**Review by renatho — APPROVED** [review]
> This issue was tricky to reproduce and fix.

I can imagine how tricky it was to figure out! Thank you for fixing it and for the amazing storytelling in the PR description! haha

I could test it and it worked for me, so I'm already approving, but I think it's worth waiting for @donnapep's answer just to make sure everything is fine.

I think the normal behavior is having the links going to the wpcomstaging.com site when they load on WP Admin instead of Calypso. When I click on Sensei LMS > Home, for example, I'm redirected to `https://test29130.wpcomstaging.com/wp-admin/admin.php?page=sensei`. Maybe it was a confusion with this behavior?
It also asks me to login again when redirecting to wpcomstaging.com, but to make sure I tested it again in wordpress.com (staging) and `trunk`, and I saw the same behavior there.

**Comment by donnapep** [comment]
Thanks @renatho and @Imran92! Please don't wait on me for approval. Renatho's approval is enough. 🙂

**Comment by obenland** [comment]
> Also had to add the delays, otherwise, I was seeing some discrepancies here and there.

Can share more details on the discrepancies you saw?
Rather than adding delays, could we move that operation into the backend and do it asynchronously?

**Comment by Imran92** [comment]
HI @obenland 👋

During my testing, a couple of times after the site setup was finished, I didn't see the 'Finish Sensei Setup' item in the checklist. It was only happening intermittently, adding the delay fixed it for me.

For the delays, I actually had an idea and I have just created [an issue](https://github.com/Automattic/wp-calypso/issues/85241) for that. I've realized that [this segment](https://github.com/Automattic/wp-calypso/blob/d7e4f48596585cd029e2a56a82cf2212374b0057/client/landing/stepper/declarative-flow/internals/steps-repository/sensei-plan/create-sensei-site.ts#L102-L116) of code was there only for setting up the variation style. We have an [API now](https://github.com/Automattic/wp-calypso/blob/d7e4f48596585cd029e2a56a82cf2212374b0057/packages/onboarding/src/cart/index.tsx#L333) to set up both the theme and the variation. I've already used the API now to setup the theme, but the variation setup didn't work for me with the API. So I'm planning to look into fixing the API under the issue I've created. Once we fix that, we'll be able to get rid of this whole segment of code and thus the associated delays.

**Comment by obenland** [comment]
Love it! See also https://github.com/Automattic/wp-calypso/issues/85170#issuecomment-1854215964 which would be helped by that

**Pull request merged**
